service/tablet_allocator.cc

```diff
@@ -1149,6 +1150,36 @@ class load_balancer {
     }
 };
 
+future<tablet_reallocation_result>
+reallocate_tablets_for_new_rf(const tablet_aware_replication_strategy* rep, schema_ptr s, token_metadata_ptr tm,
+    std::unordered_map<sstring, sstring> const& ks_options) {
+
+    auto* net_rep = dynamic_cast<const locator::network_topology_strategy*>(rep);
+
+    auto table_id = s->id();
+    tablet_map old_tablets = tm->tablets().get_tablet_map(table_id);
+
+    if (!net_rep) {
+        co_return tablet_reallocation_result{ old_tablets, tablet_reallocation_status::unknown_topology_strategy };
+    }
+
+    std::map<sstring, sstring> map_params(ks_options.cbegin(), ks_options.cend());
+    auto tablet_count = old_tablets.tablet_count();
+    locator::replication_strategy_params params{map_params, tablet_count};
+
+    auto new_rep = abstract_replication_strategy::create_replication_strategy(
+                "NetworkTopologyStrategy", params);
+
+    auto tablet_aware = new_rep->maybe_as_tablet_aware();
+
+    if (!tablet_aware) {
+        co_return tablet_reallocation_result{ old_tablets, tablet_reallocation_status::unknown_topology_strategy };
+    }
```

Comment:
I believe this check is already expressed above, but above you're just doing it in a reverse order, but that should be good enough.